x-pack/plugins/observability/public/pages/slo_edit/helpers/use_check_form_partial_validities.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { FormState, UseFormGetFieldState } from 'react-hook-form';
+import type { CreateSLOParams } from '@kbn/slo-schema';
+
+interface UseCheckFormPartialValiditiesProps {
+  getFieldState: UseFormGetFieldState<CreateSLOParams>;
+  formState: FormState<CreateSLOParams>;
+}
+
+export function useCheckFormPartialValidities({
+  getFieldState,
+  formState,
+}: UseCheckFormPartialValiditiesProps) {
+  const isDefinitionValid = (
+    [
+      'indicator.params.index',
+      'indicator.params.filter',
+      'indicator.params.good',
+      'indicator.params.total',
+    ] as const
+  ).every((field) => getFieldState(field, formState).error === undefined);
+
+  const isObjectiveValid = (
+    ['budgetingMethod', 'timeWindow.duration', 'objective.target'] as const
+  ).every((field) => getFieldState(field, formState).error === undefined);
```

Comment:
We need two validations for the Objective:
- Occurrences validation which is the one currently supported
- Timeslices validation which requires objective.timeslicesTarget and objective.timeslicesWindow.

Reply:
This is a good point. I'll create a separate issue for it so we'll not forget.

Reply:
https://github.com/elastic/kibana/issues/148254